sylt-compiler/src/typechecker.rs

```diff
@@ -1251,11 +1265,11 @@ impl TypeChecker {
 
         self.types[b].parent = Some(a);
         self.types[a].size += self.types[b].size;
-        self.types[a].constraints = self.types[a]
-            .constraints
-            .union(&self.types[b].constraints)
-            .cloned()
-            .collect();
+
+        // TODO(ed): Which span should we keep? The one closest to the top? Should we combine them?
```

Comment:
Does it matter? It is the same constraint.

Reply:
Maybe it's a good idea to proratize spans that are in the same file? I feel like this can matter, and it feels like this can stay as a todo?